File: sdks/python/pemrix/faucet_client.py
"""Faucet client for requesting test tokens."""

from dataclasses import dataclass


@dataclass
class FaucetResponse:
    success: bool
    tx_hash: str
    message: str


class FaucetClient:
    def __init__(self, base_url: str):
        self.base_url = base_url

    async def request(self, address: str, amount: str) -> FaucetResponse:
        return FaucetResponse(
            success=True,
            tx_hash="mock",
            message="mock faucet response",
        )

File: sdks/go/faucet_client.go
package pemrix

// FaucetResponse represents a faucet request response.
type FaucetResponse struct {
	Success bool   `json:"success"`
	TxHash  string `json:"tx_hash"`
	Message string `json:"message"`
}

// FaucetClient requests test tokens from a PEMRIX faucet.
type FaucetClient struct {
	BaseURL string
}

// NewFaucetClient creates a new faucet client.
func NewFaucetClient(baseURL string) *FaucetClient {
	return &FaucetClient{BaseURL: baseURL}
}

// Request submits a token request to the faucet.
func (c *FaucetClient) Request(address, amount string) (*FaucetResponse, error) {
	return &FaucetResponse{
		Success: true,
		TxHash:  "mock",
		Message: "mock faucet response",
	}, nil
}
